src/HotChocolate/Core/src/Types/Resolvers/DefaultResolverCompiler.cs

```diff
@@ -448,7 +448,8 @@ private static bool IsPureResolverResult(Type resultType)
         ParameterInfo parameter,
         IParameterExpressionBuilder[] fieldParameterExpressionBuilders)
     {
-        if (_cache.TryGetValue(parameter, out IParameterExpressionBuilder? cached))
+        if (fieldParameterExpressionBuilders.Length == 0 &&
```

Comment:
This allows us to recompile a resolver by overwriting the expression builders with locally provided ones.

Reply:
We might want to ensure that local field expression builder are not cached.